crates/core/tedge_agent/src/entity_manager/server.rs

```diff
@@ -307,23 +313,40 @@ impl EntityStoreServer {
     async fn deregister_entity(&mut self, topic_id: &EntityTopicId) -> Vec<EntityMetadata> {
         let deleted = self.entity_store.deregister_entity(topic_id);
         for entity in deleted.iter().rev() {
-            for twin_key in entity.twin_data.keys() {
-                let topic = self.mqtt_schema.topic_for(
-                    &entity.topic_id,
-                    &Channel::EntityTwinData {
-                        fragment_key: twin_key.to_string(),
-                    },
-                );
-                let clear_twin_msg = MqttMessage::new(&topic, "").with_retain();
-
-                self.publish_message(clear_twin_msg).await;
+            // Clear all retained metadata/data associated with the entity
+            match self.mqtt_connector.connection_for(&entity.topic_id).await {
```

Comment:
I would consider to spawn the whole loop in a background task (including open & close of the connection).

Reply:
Does that really help? Because, even if we spawn it as a different task, we'll still have to wait for that task to finish here, as we can't return from this `deregister` call before everything is cleared.